plugin/rewrite/rewrite.go

```diff
@@ -51,23 +51,17 @@ func (rw Rewrite) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg
 				return dns.RcodeServerFailure, fmt.Errorf("invalid name after rewrite: %s", x)
 			}
 			respRule := rule.GetResponseRule()
-			if respRule.Active == true {
+			if !rw.noRevert && respRule.Active == true {
 				wr.ResponseRewrite = true
 				wr.ResponseRules = append(wr.ResponseRules, respRule)
 			}
 			if rule.Mode() == Stop {
-				if rw.noRevert {
```

Comment:
I'm afraid this is not correct. If the last rule in a series of continued rules is not triggered (RewriteIgnored), then you will skip ResponseReverter even if some previously triggered rules had corresponding response rules.

Reply:
I do not think so. If we never go into the "stop" part, then the query will be processed by the "NextOrFailure" that is at the end of the procedure and use - the same way- the ResponseReverter.

Although I agree, would be better to show only one way to run the NextOfFailure.
Will try to propose another solution.

Reply:
Oh, I didn't see that the last `NextOrFailure()` also uses `wr`
Perhaps, you can just do `break` on Stop to avoid code duplication

Reply:
yes .. that was my though too.